File: app/src/main/java/com/example/cashflow/dataClass/City.java
package com.example.cashflow.dataClass;

public class City {
    private int id;

    private String nameCity;
    private double latitude;
    private double longitude;

    public City() {

        this.nameCity = "Bologna";
        this.latitude = 44.50;
        this.longitude = 11;
    }

    public City(String cityName, double latitude, double longitude) {
        this.nameCity = cityName;
        this.latitude = latitude;
        this.longitude = longitude;
    }

    public City(int id, String cityName, double latitude, double longitude) {
        this.id = id;
        this.nameCity = cityName;
        this.latitude = latitude;
        this.longitude = longitude;
    }

    public String getNameCity() {
        return nameCity;
    }

    public void setNameCity(String nameCity) {
        this.nameCity = nameCity;
    }

    public double getLatitude() {
        return latitude;
    }

    public void setLatitude(double latitude) {
        this.latitude = latitude;
    }

    public double getLongitude() {
        return longitude;
    }

    public void setLongitude(double longitude) {
        this.longitude = longitude;
    }

    public int getId() {
        return id;
    }


    @Override
    public String toString() {
        return "City{" +
                "nameCity='" + nameCity + '\'' +
                ", latitude=" + latitude +
                ", longitude=" + longitude +
                '}';
    }

    public String printOnApp() {
        if (nameCity != null) {
            return "Città: " + nameCity;
        } else {
            return "Città sconosciuta";
        }
    }
}

File: app/src/main/java/com/example/cashflow/dataClass/City.kt
package com.example.cashflow.dataClass

import java.io.Serializable

class City : Serializable {
    var id: Int = -1
        private set
    var nameCity: String? = null
    var latitude: Double = 0.0
    var longitude: Double = 0.0

    constructor() {
        nameCity = "Bologna"
        latitude = 44.50
        longitude = 11.0
    }

    constructor(cityName: String?, latitude: Double, longitude: Double) {
        nameCity = cityName
        this.latitude = latitude
        this.longitude = longitude
    }

    constructor(id: Int, cityName: String?, latitude: Double, longitude: Double) {
        setId(id)
        nameCity = cityName
        this.latitude = latitude
        this.longitude = longitude
    }

    fun setId(id: Int) {
        if (id > 0) {
            this.id = id
        } else {
            this.id = -1 // Oppure lascia il valore predefinito se id non è valido.
        }
    }

    override fun toString(): String {
        return "City{" +
                "id=$id" +
                ", nameCity='$nameCity'" +
                ", latitude=$latitude" +
                ", longitude=$longitude" +
                '}'
    }

    fun printOnApp(): String {
        return if (nameCity != null) {
            "Città: $nameCity"
        } else {
            "Città sconosciuta"
        }
    }
}
